simulator/tools/create_pub_sub_json_for_ui.py

```diff
@@ -0,0 +1,236 @@
+
+
+import json
+from simulator.core import protobuf_autoloader as autoloader
+import os
+import simulator.utils.constant as CONSTANTS
+from simulator.tools.common_methods import get_field_info, get_max, get_min_value, get_property_text, get_type_in_string
+
+result_data = {}
+additional_data = {}
+services = list(autoloader.get_services())
+
+
+def get_config_display_name(name):
+    return name.rsplit(".", 1)[-1]
+
+
+def get_config_name(topic):
+    return topic.rsplit('/', 1)[-1].rsplit('#', 1)[0]
+
+
+def get_pubsub(service_name):
+    output = set()
+    for item in autoloader.get_topics_by_proto_service_name(service_name):
+        status = item.split('#')[-1]
+        output.add(status)
+    return list(output)
+
+
+def get_topics_by_resource_name(resource_name, service_name):
+    topics = autoloader.get_topics_by_proto_service_name(service_name)
+    return [topic for topic in topics if topic.lower().split("#")[1] == resource_name.lower()]
+
+
+def get_pure_class_type(topic):
+    return next((pair[1] for pair in autoloader.topic_messages if topic == pair[0]), None)
+
+
+def get_ui_details(topic):
+    all_field_info = {}
+    class_type = get_pure_class_type(topic)
+    message_class = autoloader.find_message(class_type)
+    field_info = {}
+    for field_descriptor in message_class.DESCRIPTOR.fields:
+        if not (field_descriptor.type == 14 and field_descriptor.enum_type.name in ["Resource", "Resources"]):
+            field_info[field_descriptor.name] = get_field_info(field_descriptor)
+    message_name = message_class.DESCRIPTOR.name
+    all_field_info[message_name] = field_info
+    return all_field_info
+
+
+def remove_key_prefix(json_data, prefix):
+    if isinstance(json_data, dict):
+        for key, value in list(json_data.items()):  # Use list() to avoid modifying the dictionary while iterating
+            if key == "property" and value.startswith(prefix):
+                # Remove the key prefix
+                value = value[len(prefix):]
+                json_data[key] = value
+                property_text_key = key + "Text"
+                if property_text_key in json_data and json_data[property_text_key] == value:
+                    del json_data[property_text_key]  # Remove propertyText key if its value is the same as property
+            elif isinstance(value, (dict, list)):
+                remove_key_prefix(value, prefix)
+    elif isinstance(json_data, list):
+        for item in json_data:
+            if 'key' in item:
+                new_prefix = item.get('key') + "."
+                item['key'] = item['key'][len(prefix):]
+                remove_key_prefix(item, new_prefix)
+            remove_key_prefix(item, prefix)
+
+
+def extract_fields(data):
+    result = []
+    unique_ui = []
+    check = False
+    for key, value in data.items():
+        if isinstance(value, dict):
+            if 'type_field' in value and value['type_field'] not in [11, 14] and value.get('label') == 'Non-repeated':
```

Comment:
use FieldDescriptor.TYPE_MESSAGE,FieldDescriptor.TYPE_ENUM instead of 11,14

Reply:
done